benchmark/utils_sr.js

```diff
@@ -0,0 +1,172 @@
+    if (this.outHeight > this.webHeight) {
+      this.outHeight = this.webHeight;
+      this.outWidth = this.outHeight * (this.modelInfoDict.outputSize[1] / this.modelInfoDict.outputSize[0]);
+    }
+    canvasElement.setAttribute("width", this.inWidth);
+    canvasElement.setAttribute("height", this.inHeight);
+    this.outputElement.setAttribute("width", this.outWidth);
+    this.outputElement.setAttribute("height", this.outHeight);
+    canvasElement.getContext('2d').drawImage(imageElement, 0, 0, this.inWidth, this.inHeight);
+  }
+
+  async setInputOutput() {
+    this.setInputOutputUI();
+    let inWidth = this.modelInfoDict.inputSize[1];
+    let inHeight = this.modelInfoDict.inputSize[0];
+    const channels = this.modelInfoDict.inputSize[2];
+    const imageChannels = 4; // RGBA
+    const [mean, offset] = [127.5, 1];
+    this.isQuantized = this.modelInfoDict.isQuantized || false;
+    let typedArray;
+    if (this.isQuantized) {
+      typedArray = Uint8Array;
+    } else {
+      typedArray = Float32Array;
+    }
+    if (bkImageSrc === null) {
+      bkImageSrc = imageElement.src;
+    } else {
+      imageElement.src = bkImageSrc;
+    }
+    this.inputTensor = new typedArray(this.modelInfoDict.inputSize.reduce((a, b) => a * b));
+    this.outputTensor = new typedArray(this.modelInfoDict.outputSize.reduce((a, b) => a * b));
+    this.inCtx.drawImage(imageElement, 0, 0, inWidth, inHeight);
+    let pixels = this.inCtx.getImageData(0, 0, inWidth, inHeight).data;
+    for (let y = 0; y < inHeight; ++y) {
+      for (let x = 0; x < inWidth; ++x) {
+        for (let c = 0; c < channels; ++c) {
+          let value = pixels[y * inWidth * imageChannels + x * imageChannels + c];
+          this.inputTensor[y * inWidth * channels + x * channels + c] = value / mean - offset;
+        }
+      }
+    }
+  }
+
+  /**
+   * Setup model
+   * @returns {Promise<void>}
+   */
+  async setupAsync() {
+    await this.setInputOutput();
+    let backend = this.backend.replace('WebNN', 'WebML');
+    const modelName = this.modelInfoDict.modelName;
+    let loadResult = await loadModelAndLabels(this.modelInfoDict.modelFile, null);
+    let flatBuffer = new flatbuffers.ByteBuffer(loadResult.bytes);
+    let rawModel = tflite.Model.getRootAsModel(flatBuffer);
+    let kwargs = {
+      rawModel: rawModel,
+      backend: backend,
+      prefer: getPreferString()
+    };
+    this.model = new TFliteModelImporter(kwargs);
+    supportedOps = getSelectedOps();
+    await this.model.createCompiledModel();
+  }
+
+  /**
+   * Execute model
+   * @returns {Promise<void>}
+   */
+  async executeSingleAsync() {
+    let result = await this.model.compute([this.inputTensor],
+                                          [this.outputTensor]);
+    console.log(`compute result: ${result}`);
+  }
+
+  async executeAsync() {
+    let results = [];
+    for (let i = 0; i < this.iterations; i++) {
+      this.onExecuteSingle(i);
```

Comment:
Please add this code 
```javascript
await new Promise(resolve => requestAnimationFrame(resolve));
```
after 
```javascript
this.onExecuteSingle(i);
```
to update execution process status likes **Iteration: <i> / <iterations>** on UI

There is the same issue in benchmark/utils_ic.js, please help fix it toghter, thanks